src/documentation/execution-model.md

```diff
@@ -0,0 +1,163 @@
+---
+layout: default
+title: "Beam Execution Model"
+permalink: /documentation/execution-model/
+---
+
+# Apache Beam Execution Model
+
+The Beam model allows runners to execute your pipeline in different ways.
+Depending on the runner’s choices, you may observe various effects as a result.
+This page describes the effects of these choices so you can better understand
+how Beam pipelines execute.
+
+**Table of Contents:**
+* TOC
+{:toc}
+
+## Processing of elements
+
+The serialization and communication of elements between machines is one of the
+most expensive operations in a distributed execution of your pipeline. Avoiding
+this serialization may require re-processing elements after failures or may
+limit the distribution of output to other machines.
+
+The runner processes elements on many machines and may serialize elements in
+between machines for other communication and persistence reasons.
+
+### Serialization and communication
+
+The runner may serialize elements for communication or persistence.
+
+The runner may decide to transfer elements between transforms in a variety of
+ways, such as:
+
+1.  Routing elements to a worker for processing as part of a grouping operation.
+    This may involve serializing elements and sorting them by their key.
```

Comment:
s/sorting/grouping or sorting/

Specifically, let's leave how they are grouped by their serialized key open, since runners may do it differently.